### ISSUE TITLE
Travis CI build matrix

### ISSUE THREAD

**Issue by larsbrinkhoff** [forthy42/gforth#2]
If you also like to build on OS X, see this: https://docs.travis-ci.com/user/multi-os/ 
(you may have to ask support@github.com to enable this for your repo)

And you can set environment variables to different values if you want to build with different configurations.

For example, I build my Forth using Linux and OS X, `gcc -m32` and `gcc -m64`, and C and x86 targets.  That would be eight build jobs, but x86 is not supported for OS X yet, so I exclude those two:

```
os:
  - linux
  - osx
env:
  - TARGET=c M32=
  - TARGET=c M32=-m32
  - TARGET=x86 M32=
  - TARGET=x86 M32=-m32
matrix:
  exclude:
    - os: osx
      env: TARGET=x86 M32=
    - os: osx
      env: TARGET=x86 M32=-m32
```

**Comment by forthy42** [comment]
Am Donnerstag, 10. Dezember 2015, 23:39:11 schrieb Lars Brinkhoff:

> If you also like to build on OS X, see this:
> https://docstravis-cicom/user/multi-os/ (you may have to ask
> support@githubcom to enable this for your repo)
> 
> And you can set environment variables to different values if you want to
> build with different configurations

A clang build takes for ages, with libs one or two hours, due to clang taking 
an awful lot of time on register coalescing.  So unless I can use a real gcc 
on OS X to build, I'd rather not waste the resources of Travis.

Other difficulties I've, and probably can fix easily, but any help is welcome:

Build with gcc -m32 fails, probably since there is no 32 bit gcc installed.

sudo apt-get install ... won't work on OS X.  What kind of packet manager do 
they have available?  fink?  If they had fink, I could install a real gcc.

## 

Bernd Paysan
"If you want it done right, you have to do it yourself"
net2o ID: kQusJzA;7_?t=uy@X}1GWr!+0qqp_Cn176t4(dQ_
http://bernd-paysan.de/

**Comment by larsbrinkhoff** [comment]
It seems you fixed the issue with 32-bit gcc.

The supported package manager for OS X is homebrew.  I have used `brew install gforth` successfully.  libffi is brought in automatically.  I don't know if there is a gcc, or which version.

Finally, the above snippet was just an example.  You can remove `TARGET`, it's only used in my build scripts.  You don't need `matrix:` either for now.

**Comment by larsbrinkhoff** [comment]
Fixed by #5.
